src/github/client/interceptors.ts

```diff
@@ -84,10 +85,10 @@ export const handleFailedRequest = (logger: Logger) =>
 			const errorMessage = `Error executing Axios Request ` + error.message;
 
 			const rateLimitResetHeaderValue:string = response.headers?.["x-ratelimit-reset"];
-			const rateLimitRemainingHeaderValue:string = response.headers?.["x-ratelimit-reset"];
-			if(response.headers?.["x-ratelimit-remaining"] == "0" && rateLimitResetHeaderValue) {
+			const rateLimitRemainingHeaderValue:string = response.headers?.["x-ratelimit-remaining"];
+			if(status === 403 && rateLimitRemainingHeaderValue == "0") {
 				logger.warn({ err: error }, "Rate limiting error");
-				const rateLimitReset: number = parseInt(rateLimitResetHeaderValue);
+				const rateLimitReset: number = parseInt(rateLimitResetHeaderValue) || Date.now() / 1000 + ONE_HOUR_IN_SECONDS;
```

Comment:
Falling back to rate limit reset one hour from now if it hasn't been specified for some reason.

Reply:
Would it be an exception from parseInt function if rateLimitResetHeaderValue is undefined?

Reply:
no, `parseInt()` returns `NaN` in that case, and that is `falsy`, so we fall back to the value on the right side of the `||`.